Track dependency of `skipped_as_script` cells

disberd opened the pull request:
Following the discussion in https://github.com/fonsp/Pluto.jl/discussions/2175

This is a first draft PR to also comment out in the file cells that depend on ones that are directly flagged as _disabled in file_.

Implementation was quite straightforward following the similar dependency on disabled cells, but an update trigger is also needed outside of the `reactive_run_core!` function as this flag might create a notebook update without causing cells running.

Didn't put any styling difference in the cells visuals but modified the click tooltip to differentiate between directly and indirectly skipped cells.

disberd commented:
@greimel Since you expressed interest, let me know if you find issues if trying this out!

greimel commented:
Great! I've just tried and it doesn't seem to work. Could it be that the flag is still there?

I don't know anything about HTML and JS, but this doesn't seem to work.

https://user-images.githubusercontent.com/6280307/174314935-627141d0-8800-42ca-b80f-0d4ec81bfd15.mov

Pangoraw commented:
try this to enable it:
```html
<script>
if (window.localStorage.getItem("ENABLE_SKIP_AS_SCRIPT") !== "true") {
    window.PLUTO_TOGGLE_SKIP_AS_SCRIPT()
}
</script>
```
 as your current code is toggling then reloading then re-toggling, etc...

greimel commented:
I've played around with it a bit and it seems to work well. Unfortunately it just comes after finishing my course preparations. Otherwise I could have battle-tested it a bit more :-)

greimel commented:
It would be nice if there was a way to jump from the dependent cell to the one that's been skipped. This would be similar to `disable`

![image](https://user-images.githubusercontent.com/6280307/174340412-f555aa57-122f-43c2-ab32-fc8cb1c8c153.png)

You can click on the arrow to jump to the cell.

disberd commented:
Indeed I didn't notice you could do so with disabled cells, implementation it should be straightforward but it would be nice to have a button to trigger this.

One idea could be to re-use the context menu button. Making it only go to the parent cell for a cell that is indirectly disabled

Pangoraw commented:
I changed the styling of the marker so that the tooltip is only on the left border and not on the entire cell output (which meant that the cell output could not be interacted with).

![image](https://user-images.githubusercontent.com/9824244/174472734-c53f3907-5193-4422-b37f-d275f80477b1.png)

What do you think ?

Pangoraw commented on src/webserver/Dynamic.jl:
```diff
@@ -111,6 +111,7 @@ function notebook_to_js(notebook::Notebook)
                 "code" => cell.code,
                 "code_folded" => cell.code_folded,
                 "metadata" => cell.metadata,
+                "depends_on_skipped_cells" => cell.depends_on_skipped_cells,
```
I think this makes more sense in `"cell_results"` like `"depends_on_disabled_cells"` meaning that you won't need to initialize it everywhere in the frontend.

disberd replied:
Yeah I didn't realize you would have to initialize it before going into the typescript error.
I put it into the cell_inputs as it seemed more appropriate there logically (I see it more as something related to the cell input rather than the cell result).
But indeed it's probably easier to make it part of the cell results

Pangoraw replied:
> I see it more as something related to the cell input rather than the cell result

The rationale is that cell inputs are set by the user whereas cell results can be determined from all inputs (code, topology, etc...).

disberd replied:
That also makes sense! Just change the code accordingly

disberd commented:
I implemented a few changes:

I added a stripes to the visual indicator of indirectly skipped cells to differentiate them and changed the text of the tooltip a bit to also provide a jump to the upstream cell causing the indirect skipping:
![image](https://user-images.githubusercontent.com/12846528/174731839-fdeaa4c5-dcb2-420c-9491-3ffa40d44b4f.png)

To provide the jump functionality, I reused the code of disabled cells so I moved the relevant functions from RunArea.js into Cell.js and made them generic instead of specific for the `disabled` metadata.
Maybe @lungben have a look as well. 
Both disabled and skipped behavior seem to work correctly but probably the javascript side on Cell.js can be made cleaner, @Pangoraw or @pankgeorg can provide feedback here.

Pangoraw reviewed (approved):
Looks good to me! I changed the appearance of dependents on skip as script to use a lighter/darker background instead of the gradient which I think looked too much like the cell running indicator, what do you think ?

disberd commented:
@Pangoraw that is also OK with me. From my point of view this is ready to be merged

greimel commented:
It seems to me that the feature is still behind a flag. Shouldn't the flag be removed?

disberd commented:
You are right! I just removed the flag in the last commit

Pangoraw commented:
Let's merge when tests pass! It would be great to have a wiki page/sample notebook about this feature and cell disabling.

fonsp commented:
This is great! Just used it in https://github.com/JuliaPluto/PlutoUI.jl/pull/231 , it makes it so easy to disable all the test cells in the notebook file! Before, with the `@skip_as_script` macro I was just too lazy to put the macro in front of every cell.

In one of the notebooks, I do `using PlutoUI` to get `Slider`, `Button`, etc to test the notebook with, and I can just disable the `using PlutoUI` cell, and it will automatically disable everything I need! I think that would also be really useful with `using PlutoTest`
